### PR TITLE
Get last index timestamp

### PR DESCRIPTION
**Pull request by Tsytrytskyi**
Summary
---------
url --request GET   --url http://127.0.0.1:5000/m/index/pdq/status   --header 'Content-Type: application/json' 
{
  "timestamp": 1696449928
}

curl --request GET   --url http://127.0.0.1:5000/m/index/pdqa/status   --header 'Content-Type: application/json' 
{
  "message": "The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again."
}

Test Plan
---------

### CODE REVIEW

File: open-media-match/src/OpenMediaMatch/blueprints/curation.py [hunk 1]
```diff
@@ -21,6 +21,13 @@ def banks_index():
     storage = persistence.get_storage()
     return list(storage.get_banks().values())
 
+@bp.route("/banks_with_content_count", methods=["GET"])
+@utils.abort_to_json
+def banks_with_content_count():
+    storage = persistence.get_storage()
+    print(storage.get_banks_with_content_count())
+    return storage.get_banks_with_content_count()
```
Comment (Dcallies) [inline open-media-match/src/OpenMediaMatch/blueprints/curation.py:29]: blocking: Remove this one, we'll use the one in matching

File: open-media-match/src/OpenMediaMatch/blueprints/matching.py [hunk 2]
```diff
@@ -106,12 +107,7 @@ def lookup():
...
 
 
-@bp.route("/index/status")
-def index_status():
-    """
-    Input:
-     * Signal type (hash type)
-    Output:
-     * Time of last index build
-    """
```
Comment (Dcallies) [inline open-media-match/src/OpenMediaMatch/blueprints/matching.py:116]: blocking: Bring back this docstring or something like it

File: open-media-match/src/OpenMediaMatch/blueprints/matching.py [hunk 3]
```diff
@@ -106,12 +107,7 @@ def lookup():
...
-    abort(501)  # Unimplemented
+@bp.route("/index/<index_type_name>/status")
+def index_status(index_type_name):
```
Comment (Dcallies) [inline open-media-match/src/OpenMediaMatch/blueprints/matching.py:111]: add a type of str to index_type_name for mypy

```
index_status(index_type_name: str)
```

File: open-media-match/src/OpenMediaMatch/templates/bank.html.j2 [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    </head>
+    <body>
+        <h3> Bank Names </h3>
+        {% for b in bankList %}
+            <tr>
+                <td> <li> {{b['name']}} </li> </td>
+            </tr>
+        {% endfor %}
+    </body>
+</html>
```
Comment (Dcallies) [inline open-media-match/src/OpenMediaMatch/templates/bank.html.j2:15]: blocking: Remove this

**Review by Dcallies — CHANGES_REQUESTED** [review]
Some blocking comments, also check out the CI failures.

File: open-media-match/src/OpenMediaMatch/blueprints/matching.py (hunk 3 above)
Comment (Tsytrytskyi) [reply open-media-match/src/OpenMediaMatch/blueprints/matching.py]: Done

File: open-media-match/src/OpenMediaMatch/templates/bank.html.j2 (hunk 4 above)
Comment (Tsytrytskyi) [reply open-media-match/src/OpenMediaMatch/templates/bank.html.j2]: Removed.

File: open-media-match/src/OpenMediaMatch/blueprints/matching.py (hunk 2 above)
Comment (Tsytrytskyi) [reply open-media-match/src/OpenMediaMatch/blueprints/matching.py]: Restored

File: open-media-match/src/OpenMediaMatch/blueprints/matching.py [hunk 5]
```diff
@@ -115,3 +116,14 @@ def index_status():
...
     """
     abort(501)  # Unimplemented
+
+
+@bp.route("/index/<index_type_name>/status")
+@abort_to_json
+def index_status_by_type(index_type_name: str):
+    storage = persistence.get_storage()
+    timestamp = storage.get_last_signal_build_timestamp(index_type_name)
+    if timestamp:
```
Comment (Dcallies) [inline open-media-match/src/OpenMediaMatch/blueprints/matching.py:126]: Should be if timestamp is not None - 0 might trigger this otherwise

**Review by Dcallies — APPROVED** [review]

**Pull request merged**